shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -155,6 +146,22 @@ public boolean onTouchEvent(@NonNull MotionEvent event, @NonNull Matrix transfor
         !updateForSinglePointer
             && (maskedAction == MotionEvent.ACTION_UP
                 || maskedAction == MotionEvent.ACTION_POINTER_UP);
+
+    int deviceType = getPointerDeviceTypeForToolType(event.getToolType(event.getActionIndex()));
+    boolean shouldRemovePointer =
+        updateForMultiplePointers && (deviceType == PointerDeviceKind.TOUCH);
```

Comment:
Can you illustrate why the synthesis is only needed for `updateForMultiplePointers`?

Reply:
Oh I see, is it because `updateForMultiplePointers` is basically this-event-is-up?

Reply:
Longwinded ish answer: I think the naming makes it a little confusing, but my understanding is that if we take the definition of `updateForMultiplePointers`
```dart
boolean updateForMultiplePointers =
        !updateForSinglePointer
            && (maskedAction == MotionEvent.ACTION_UP
                || maskedAction == MotionEvent.ACTION_POINTER_UP);
```
which is equivalent to 
```dart
boolean firstCondition = !(MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN);
boolean secondCondition = (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP);
boolean updateForMultiplePointers = firstCondition && secondCondition;
```
If `secondCondition` is true, then `firstCondition` is also guaranteed to be true. And if `secondCondition` is false, then `updateForMultiplePointers` is going to be false, regardless of what `firstCondition` is.
So in reality `updateForMultiplePointers` is actually just
```dart
boolean updateForMultiplePointers = secondCondition;
// or
boolean updateForMultiplePointers = (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP);
```

Which is exactly the case we are looking to remove pointers for - on `ACTION_UP` and `ACTION_POINTER_UP`, aka when a finger leaves the screen.

Reply:
> Oh I see, is it because `updateForMultiplePointers` is basically this-event-is-up?

Yeah this comment hadn't loaded for me when I wrote the longwinded answer above haha